whitepapers/tickets/SHARE-822.md

```diff
@@ -0,0 +1,232 @@
+# [SHARE-822] Custom Taxonomies
+
+
+## Goals:
+ - Support OSF's Custom Taxonomies requirements
+ - Have a solution that can be used by any parties interested in enhancing SHARE
+ - Migrate the existing Subjecting system to this new solution
+ - Continue to use the Bepress taxonomy as SHARE's central taxonomy
+
+
+## Reference
+The following values will be refered to for examples throughout this document
+
+```
+Taxonomy X:
+  Subject Alpha -> Bepress A
+    Subject Bravo -> Bepress B
+      Subject Charlie -> Bepress C
+    Subject Delta -> Bepress B
+  Subject Echo -> Bepress D
+```
+
+
+
+## Changes to OSF
+
+Changes are to be done by the OSF team.
+
+OSF's serialization of `Subject`s and `ThroughSubject`s in the `on_preprint` updated method will neeed to be modified.
+
+
+### Subject
+
+```javascript
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+
+  "properties": {
+    "@id": {"type": "string"},  // Unchanged
+    "@type": {"enum": ["subject"] },  // Unchanged
+    "name": {"type": "string"}, // Unchanged
+    "parent": {"type": {"$ref": "#/definitions/reference"}},  // Unchanged
+
+    "central_synonyms": {
+      "type": {"$ref": "#/definitions/reference"}
+      "default": null,
+      "description": "The central (Currently Bepress) subject that this subject is equivalent to."
+      // Nullable right now. It might make sense to make this a self reference for Bepress Subject
```

Comment:
We should probably decide to either make `null` semantically significant going forward, or be self-referential from the get-go.